maven-java/kalix-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml

```diff
@@ -23,7 +23,7 @@
     <maven.build.timestamp.format>yyyyMMddHHmmss</maven.build.timestamp.format>
     <mainClass>${package}.Main</mainClass>
 
-    <jdk.target>17</jdk.target>
+    <jdk.target>21</jdk.target>
```

Comment:
Unfortunately that's not enough:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project java-spring-transfer-workflow-compensation: Compilation failure
[ERROR] /Users/andrzej/projects/kalix/kalix-jvm-sdk/samples/java-spring-transfer-workflow-compensation/src/main/java/com/example/transfer/TransferWorkflow.java:[127,18] patterns in switch statements are not supported in -source 17
[ERROR]   (use -source 21 or higher to enable patterns in switch statements)
```

It is because we inherit this config from spring (even after bumping from 3.1.6 to 3.2.0):
```
<modelVersion>4.0.0</modelVersion>
  <parent>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-dependencies</artifactId>
    <version>3.2.0</version>
  </parent>
  <artifactId>spring-boot-starter-parent</artifactId>
  <packaging>pom</packaging>
  <name>spring-boot-starter-parent</name>
  <description>Parent pom providing dependency and plugin management for applications built with Maven</description>
  <properties>
    <java.version>17</java.version>
    <resource.delimiter>@</resource.delimiter>
    <maven.compiler.release>${java.version}</maven.compiler.release>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
  </properties>
```
and then this config is used by the `maven-compiler-plugin`: https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-release.html

We can solve this in 2 ways:
1. either change the compiler plugin from:
```
<artifactId>maven-compiler-plugin</artifactId>
        <configuration>
          <source>${jdk.target}</source>
          <target>${jdk.target}</target>
```
to
```
<artifactId>maven-compiler-plugin</artifactId>
        <configuration>
          <release>${jdk.target}</release>
```
2. or to simply override `java.version` and remove `jdk.target` property